mod/account/set/right/password.rb

```diff
@@ -15,14 +22,45 @@ def ok_to_read
   self.content = Auth.encrypt content, salt
 
   # errors.add :password, 'need a valid salt'
-  # turns out we have a lot of existing account without a salt.
+  # turns out we have a lot of existing accounts without a salt.
   # not sure when that broke??
 end
 
-event :validate_password, :validate, on: :save do
-  return if content.length > 3
+event :validate_password_length, :validate, on: :save do
+  min_pw_length = Cardio.config.account_password_length
+  return if content.length >= min_pw_length
+
+  errors.add :password, t(:account_password_length, num_char: min_pw_length)
+end
+
+def check_password_regex char_types, regex_hash, password
+  pw_requirements = char_types.each_with_object([]) do |char_type, pw_requirements|
```

Comment:
what codeclimate doesn't like is that you're using the same variable name for the array inside the iteration and the one outside. That makes it harder to understand what the code is doing.

Reply:
It is intended to be the same variable as in the outer scope.

Reply:
it's the same value, but it's two different variables.